File: app-presentation/src/main/java/com/teewhydope/app/presentation/login/mapper/LoginResponseDomainToPresentationModelMapper.kt
package com.teewhydope.app.presentation.login.mapper

import com.teewhydope.app.domain.authentication.model.LoginResponseDomainModel
import com.teewhydope.app.presentation.common.mapper.DomainToPresentationMapper
import com.teewhydope.app.presentation.login.model.LoginResponsePresentationModel

class LoginResponseDomainToPresentationModelMapper(
    private val userDataDomainToPresentationModelMapper: UserDataDomainToPresentationModelMapper,
    private val smartSaverTransactionDomainToPresentationModelMapper: SmartSaverTransactionDomainToPresentationModelMapper,
) :
    DomainToPresentationMapper<LoginResponseDomainModel, LoginResponsePresentationModel>() {
    override fun map(input: LoginResponseDomainModel) = LoginResponsePresentationModel(
        userData = userDataDomainToPresentationModelMapper.toPresentation(input.userData),
        smartSaverTransactions = input.smartSaverTransactions.map {
            smartSaverTransactionDomainToPresentationModelMapper.toPresentation(it)
        }
    )
}

File: iosApp/iosApp/Presentation/Login/Mapper/LoginResponseDomainToPresentationModelMapper.swift
//
//  LoginResponseDomainToPresentationModelMapper.swift
//  iosApp
//
//  Created by Animasahun Ibrahim on 17/03/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import IosData

class LoginResponseDomainToPresentationModelMapper: DomainToPresentationMapper{
    typealias T = LoginResponseDomainModel
    
    typealias U = LoginResponsePresentationModel
    
    private let userDataDomainToPresentationModelMapper: UserDataDomainToPresentationModelMapper
    private let smartSaverTransactionDomainToPresentationModelMapper: SmartSaverTransactionDomainToPresentationModelMapper
    
    init(
        userDataDomainToPresentationModelMapper: UserDataDomainToPresentationModelMapper,
        smartSaverTransactionDomainToPresentationModelMapper: SmartSaverTransactionDomainToPresentationModelMapper
    ) {
        self
            .userDataDomainToPresentationModelMapper = userDataDomainToPresentationModelMapper
        self
            .smartSaverTransactionDomainToPresentationModelMapper = smartSaverTransactionDomainToPresentationModelMapper
    }
    
    func toPresentation(input: LoginResponseDomainModel) -> LoginResponsePresentationModel {
        
        return .init(
            userData: userDataDomainToPresentationModelMapper.toPresentation(input: input.userData),
            smartSaverTransactions: input.smartSaverTransactions.map {
                smartSaverTransactionDomainToPresentationModelMapper
                    .toPresentation(input: $0)
            }
        )
    }
    
}
